emacs_erlang_yaemep_support.erl

```diff
@@ -92,6 +94,206 @@ erlang_project_dir(ErlangFilePath) ->
+                {ok, #file_info{type=directory}} ->
+                    case is_recursive(Pattern) of
+                        true ->
+                            {Visited2, Acc2} = traverse_fs(Path, Pattern, RT,
+                                                           Visited1, Acc),
+                            tr_fs_1(Rest, Pattern, RT, Visited2, Acc2);
+                        false ->
+                            tr_fs_1(Rest, Pattern, RT, Visited1, Acc)
+                    end;
+                {ok, #file_info{type=symlink}} ->
+                    case file:read_link(Path) of
+                        {ok, SymlinkTarget} ->
+                            %% Find the symlink dest, regardless of if it is
+                            %% relative or absolute. Use the fact
+                            %% that filename:join(Abs1, Abs2) -> Abs2
+                            CurrDir = filename:dirname(Path),
+                            TargetPath = filename:join(CurrDir, SymlinkTarget),
+                            CPath = canonicalize_path(TargetPath),
+                            tr_fs_1([CPath | Rest], Pattern, RT, Visited1, Acc);
+                        {error, _} ->
+                            tr_fs_1(Rest, Pattern, RT, Visited1, Acc)
+                    end;
+                {ok, _X} ->
+                    tr_fs_1(Rest, Pattern, RT, Visited, Acc);
+                {error, _X} ->
+                    tr_fs_1(Rest, Pattern, RT, Visited, Acc)
+            end
+    end;
+tr_fs_1([], _Pattern, _RT, Visited, Acc) ->
+    {Visited, Acc}.
+
+new_set() ->
+    sets:new().
+
+add_element(Elem, Set) ->
+    sets:add_element(Elem, Set).
+
+is_in_set(Elem, Set) ->
+    sets:is_element(Elem, Set).
+
+-spec analyze_wildcard(string()) -> {TopDir, FilePattern} when
+      TopDir :: string(),
+      FilePattern :: BasePattern | {recursively, BasePattern},
+      BasePattern :: {extensions, [string()]} | % with leading dots
+                     {exact_base, string()}.
+analyze_wildcard(Wildcard) ->
+    case lists:reverse(filename:split(Wildcard)) of
+        [BasePattern] ->
+            Cwd = safe_cwd(),
+            RT = mk_safe_path_unprefixer(Cwd),
+            {Cwd, analyze_base_pattern(BasePattern), RT};
+        [BasePattern, "**"] ->
+            Cwd = safe_cwd(),
+            RT = mk_safe_path_unprefixer(Cwd),
+            {Cwd, {recursively, analyze_base_pattern(BasePattern)}, RT};
+        [BasePattern, "**" | TopDirReversed] ->
+            TopDir = filename:join(lists:reverse(TopDirReversed)),
+            RT = fun(X) -> X end,
+            {TopDir, {recursively, analyze_base_pattern(BasePattern)}, RT};
+        [BasePattern | TopDirReversed] ->
+            TopDir = filename:join(lists:reverse(TopDirReversed)),
+            RT = fun(X) -> X end,
+            {TopDir, analyze_base_pattern(BasePattern), RT}
+    end.
+
+safe_cwd() ->
+    case file:get_cwd() of
+        {ok, Cwd}  -> Cwd;
+        {error, _} -> "."
+    end.
```

Comment:
Is this function ever called by YAEMEP or is it only intended for testing of `find_files`? I think the path that `file:get_cwd()`will be whatever the current working directory is for emacs.

Reply:
Yes, good point, this makes the code much simpler. Done.